libresapi/src/api/FileSharingHandler.h

```diff
@@ -48,10 +55,13 @@ class FileSharingHandler: public ResourceRouter
 
 	void handleDownload(Request& req, Response& resp);
 
-	StateToken mStateToken;
+	StateToken mLocalDirStateToken;
+	StateToken mRemoteDirStateToken;
 	StateTokenServer* mStateTokenServer;
+	RsMutex mMtx;
```

Comment:
Here comments explaining the usage of each token and the mutex usage would be helful